[river20s] WEEK 02 solutions

river20s opened the pull request:
## 답안 제출 문제

- [x] #218 
- [x] #230 
- [x] #239 
- [ ] #241 
- [ ] #251 

## 작성자 체크 리스트

- [x] 우측 메뉴에서 PR을 **Projects**에 추가해주세요.
- [x] **Projects**의 오른쪽 버튼(▼)을 눌러 확장한 뒤, **Week**를 현재 주차로 설정해주세요.
- [ ] 문제를 모두 푸시면 프로젝트에서 **Status**를 `In Review`로 설정해주세요.
- [ ] 코드 검토자 1분 이상으로부터 승인을 받으셨다면 PR을 병합해주세요.

## 검토자 체크 리스트

> [!IMPORTANT]
> 본인 답안 제출 뿐만 아니라 다른 분 PR 하나 이상을 반드시 검토를 해주셔야 합니다!

- [x] 바로 이전에 올라온 PR에 본인을 코드 리뷰어로 추가해주세요.
- [x] 본인이 검토해야하는 PR의 답안 코드에 피드백을 주세요.
- [x] 토요일 전까지 PR을 병합할 수 있도록 승인해주세요.

river20s committed:
feat: Add empty file river20s.py

river20s committed:
Remove river20s.py

river20s committed:
feat: add solution to Valid Anagram #218

hestia-park commented on valid-anagram/river20s.py:
```diff
@@ -0,0 +1,57 @@
+        if sdict == tdict:
+            return True
+        else:
+            return False
```
저는 counter만 사용해서 풀었는데 딕셔너리 접근법도 좋은 것 같습니다! 마지막에 개행문자만 추가해주시면 될 것 같아요!

river20s committed:
feat: add solution to Climbing Stairs

river20s committed:
feat: linelint를 위한 마지막 줄 개행 문자
추가

river20s committed:
refactor: Improve algorithm performance with dynamic programming

river20s committed:
feat: Add Solution to Product of Array Except Self #239

river20s committed:
feat: linelint를 위한 마지막 줄 개행 문자 추가

river20s committed:
refactor: Optimize space usage in productExceptSelf solution

- Reuse output array for prefix cumulative product computation
- Utilize a single variable for suffix cumulative product to update output array in-place
- Achieve O(n) time and O(1) extra space complexity
